src/Core/Variables/StandardVariableProvider.php

```diff
@@ -356,20 +356,27 @@ protected function detectAccessor($subject, $propertyName)
         }
         if (is_object($subject)) {
             $className = $this->populateGettersByClassName($subject);
-            $upperCasePropertyName = ucfirst($propertyName);
-            $getter = 'get' . $upperCasePropertyName;
+            $lowerCasePropertyName = strtolower($propertyName);
+            $getter = 'get' . $lowerCasePropertyName;
             if (static::$gettersByClassName[$className][$getter] ?? false) {
                 return self::ACCESSOR_GETTER;
             }
             if ($this->isExtractableThroughAsserter($subject, $propertyName)) {
                 return self::ACCESSOR_ASSERTER;
             }
-            if (property_exists($subject, $propertyName)) {
+            $propertyNames = array_keys(get_object_vars($subject));
+            // __get() magic methods are considered as public property
+            if (in_array($propertyName, $propertyNames, true) || method_exists($subject, '__get')) {
```

Comment:
Technically, adding support for `__get` changes the behavior again because now, a class which implements `__get` will never be able to support `__call`. There is also the question if (virtual) public properties should take priority over getter methods, which would also change the behavior.